### ISSUE TITLE
Add usage instructions

### ISSUE THREAD

**Issue by MPV** [startersclan/docker-sourceservers#1]
Hi,
This looks promising — would you be open to adding some usage instructions?

**Comment by joeltimothyoh** [comment]
Hi @MPV. I've opened #2 that addresses usage instructions and some important points to take note of concerning the game images. You can view the readme file directly rather than the diffs if that helps. Have a look and let me know your thoughts.

**Comment by joeltimothyoh** [comment]
Greetings @MPV. The project's readme has been updated with changes from #2. Do let me know if you still have any doubts on how to use the game images.

**Comment by MPV** [comment]
Wow, beautiful (and detailed, yet relevant)! ❤️ 

I agree that we could close this — final remark/question: how/where would you recommend mounting `server.cfg` file(s) for easily using them? Please do excuse my novel experience in srcds hosting. 😊

**Comment by joeltimothyoh** [comment]
If it's a barebone server, you might be able to get away with just mounting a `server.cfg`. The challenge comes when you have multiple servers running different game modes, each of which having their own custom files.

If git is used source-control server custom files such as `server.cfg`, they can be added to the container at runtime using init containers. However, if `server.cfg` is going to be source-controlled, it should not contain Cvars with secrets within it. A solution is to have a baseline set of Cvars specified within `server.cfg` for each game mode or server that will be hosted, depending on your setup. All other cvars whose values differ across server instances will be overridden using one or more config override files containing Cvars and their override values, each of whose names based on the relative path of `server.cfg` would then have to be exist within `server.cfg` in the format of `exec myoverridefile.cfg`, usually placed at the end of `server.cfg` as the file is executed from top to bottom with later Cvars and values taking effect.

Depending on the game and mod (e.g. `sourcemod`), `server.cfg` may not always load on every map change, so you'll need to know the game and mod well. Also, games such as `csgo` have game mode Cvar config files such as `gamemode_competitive_server.cfg`; mods such as `sourcemod` have `sourcemod.cfg`; and some mod plugins have their own Cvar config files. Not only this, but mods like `sourcemod` also allows overriding of certain *locked* game Cvar values which can be overridden via specifying Cvars in the format e.g. `sm_cvar sv_maxcmdrate 128`, which can be placed place within `server.cfg` if its a core setting that is going to be used across all your servers.

Basically you'll have to know the sequence of how the Cvars are read, and how to best organize them that allows Cvars to be overridden if required, and in such a way that is possible via container orchestration tools such as Kubernetes. You'll also have to take into account updates to your custom files and/or overrides, and how to deploy the changes continuously and in a repeatable fashion without compromising on the uptime and reliability of servers.

@MPV

**Comment by MPV** [comment]
Thanks @joeltimothyoh — that makes sense.
I hadn't hosted a CS-server since 2004 so I had forgotten the intricacies of it's server configurations. 

Closing the issue, thanks again.
